qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -133,6 +141,46 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
 		return CompletableFuture.completedFuture(codeActions);
 	}
 
+	private void doCodeActionToCreateProperty(Template template, Diagnostic diagnostic, QuteErrorCode errorCode,
+			List<CodeAction> codeActions, QuteTemplateGenerateMissingJavaMember resolver, SharedSettings settings) {
+		JsonObject data = (JsonObject) diagnostic.getData();
+		String missingProperty = data.get("property").getAsString();
+		String resolvedType = data.get("signature").getAsString();
+		String propertyCapitalized = missingProperty.substring(0, 1).toUpperCase() + missingProperty.substring(1);
+
+		String projectUri = template.getProjectUri();
+
+		GenerateMissingJavaMemberParams publicFieldParams = new GenerateMissingJavaMemberParams(MemberType.Field,
+				missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams getterParams = new GenerateMissingJavaMemberParams(MemberType.Getter,
+				missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams templateExtensionParams = new GenerateMissingJavaMemberParams(
+				MemberType.TemplateExtension, missingProperty, resolvedType, projectUri);
+
+		CodeAction createPublicField = createCodeActionWithData(
+				String.format("Create public field \"%s\" in %s", missingProperty, resolvedType), publicFieldParams,
+				Collections.singletonList(diagnostic));
+		CodeAction createGetter = createCodeActionWithData(
+				String.format("Create getter \"get%s()\" in %s", propertyCapitalized, resolvedType), getterParams,
+				Collections.singletonList(diagnostic));
+		CodeAction createTemplateExtension = createCodeActionWithData(String
+				.format("Create template extension \"%s()\" in detected template extensions class", missingProperty),
+				templateExtensionParams, Collections.singletonList(diagnostic));
+
+		try {
+			createPublicField.setEdit(resolver.resolveTemplateCodeAction(publicFieldParams).get());

```

Comment:
Using .get could block the thread.  I would try to use .getNow(null) by checking if result != null

Reply:
I'll see if there is a way I can do this without blocking. Unfortunately, I can't use `.getNow(null)`, since the result will always be `null`.

Reply:
I tried a slightly different method to prevent blocking in QuteCodeActions. When you get a change, please let me know what you think about it.

Reply:
Yes it is better thanks. But I think we should update the edit in à resolve code action. We can do that in an another pr